network/default.go

```diff
@@ -897,6 +963,80 @@ func (n *network) processNetChan(listener tunnel.Listener) {
 				default:
 					// don't block here
 				}
+			case "sync":
+				// record the timestamp of the message receipt
+				now := time.Now()
+
+				pbNetSync := &pbNet.Sync{}
+				if err := proto.Unmarshal(m.msg.Body, pbNetSync); err != nil {
+					log.Debugf("Network tunnel [%s] sync unmarshal error: %v", NetworkChannel, err)
+					continue
+				}
+
+				// don't process your own messages
+				if pbNetSync.Peer.Node.Id == n.options.Id {
+					continue
+				}
+
+				log.Debugf("Network received sync message from: %s", pbNetSync.Peer.Node.Id)
+
+				peer := &node{
+					id:      pbNetSync.Peer.Node.Id,
+					address: pbNetSync.Peer.Node.Address,
+					link:    m.msg.Header["Micro-Link"],
+					peers:   make(map[string]*node),
+					status: &status{
+						err: &nerr{
```

Comment:
probably want a method to clean up creation of this status like `makeStatus(peer)`

Reply:
Yeah -- lots of this stuff needs to be cleaned up, not just this part 😅